torch_geometric/transforms/SKLearnPreprocessTransform.py

```diff
@@ -0,0 +1,40 @@
+from typing import List, Union
+
+import torch
+
+from torch_geometric.data import Data, HeteroData
+from torch_geometric.data.datapipes import functional_transform
+from torch_geometric.transforms import BaseTransform
+
+
+@functional_transform('sklearn_preprocess')
+class SKLearnPreprocessTransform(BaseTransform):
+    r"""Column-normalizes the attributes given in :obj:`attrs` to sum-up to standardize or scale them.
+
+    Mainly useful for edges, if the scalar for the collective data edges is stored
+    as data[(node1,to,node2)].edge_scaler = scaler
+
+    where the scaler can be StandardScaler or MinMaxScaler from scikit-learn that
+    has been fit to the total dataset
+
+    Args:
+        attrs (List[str]): The names of attributes to normalize.
+            (default: :obj:`["edge_attr"]`)
+    """
+    def __init__(self, attrs: List[str] = ["edge_attr"]):
+        self.attrs = attrs
+
+    def __call__(self, data: Union[Data, HeteroData]):
+        for edge, store in list(zip(data.edge_types, data.edge_stores)):
```

Comment:
In my current implementation, I had 2 scalers for 2 edge_types , so i add them like `data[edge].edge_scaler = ...` similar to how we have edge_index, edge_attr and edge_label.  Then I access it in the call, since the scaler needs to be fit on the complete data(multiple graphs like Dataset(30)) and not just data[0] which is just one graph. However, I think this can be a post processing step, where the transform takes in a Data/HeteroData object and calculates the scaler within the transform call instead of it being added to the data object